src/test/java/org/highj/util/Gen.java

```diff
@@ -0,0 +1,29 @@
+package org.highj.util;
+
+import org.highj.data.List;
+import org.highj.data.Stream;
+
+import java.util.Random;
+import java.util.function.Function;
+
+public interface Gen<A> {
+
+    Random rnd = new Random();
```

Comment:
`java.util.Random` is mutable. I think you should avoid `java.util.Random` and then use immutable random number generator and State monad if you want to generate arbitrary functions.
see
- https://github.com/rickynils/scalacheck/issues/136
- https://github.com/rickynils/scalacheck/pull/171